client.go

```diff
@@ -133,13 +133,18 @@ func (c *Client) Call(params, resp interface{}) error {
 // Any error that c.UnmarshalError or c.Doer returns will not
 // have its cause masked.
 //
-// The request URL field is changed to the actual URL used.
+// If c.BaseURL is not empty hten the URL in the request is expected to
```

Comment:
I wonder about something more like:

// The URL in req.URL is interpreted relative to c.BaseURL - if it
// contains a host part, then c.BaseURL will be ignored,
// otherwise it will be appended to c.BaseURL.

?